src/components/ContentHighlights/HighlightStepper/ContentHighlightStepper.jsx

```diff
@@ -89,19 +86,116 @@ const ContentHighlightStepper = ({ enterpriseId }) => {
         addHighlightSet: true,
       });
       closeStepperModal();
+      const handlePublishTrackEvent = () => {
+        const trackInfo = {
+          stepper_modal: {
+            current_step: currentStep,
+            current_step_position: steps.indexOf(currentStep) + 1,
+          },
+          created_highligh_response_result_data: {
+            card_image_url: transformedHighlightSet.cardImageUrl,
+            is_published: transformedHighlightSet.isPublished,
+            highlight_set_title: transformedHighlightSet.title,
+            highlight_set_uuid: transformedHighlightSet.uuid,
+            highlighted_content_uuids: transformedHighlightSet.highlightedContentUuids,
+          },
+        };
+        sendEnterpriseTrackEvent(
+          enterpriseId,
+          `${EVENT_NAMES.CONTENT_HIGHLIGHTS.STEPPER_STEP_CONFIRM_CONTENT}.publish.clicked`,
+          trackInfo,
+        );
+      };
+      handlePublishTrackEvent();
     } catch (error) {
       logError(error);
     } finally {
       setIsPublishing(false);
     }
   };
+  const handleNext = () => {
```

Comment:
Rather than having one function for all possible "Next" clicks, I think it might be worth having separate function for each distinct "Next" click to simplify this function and remove the conditional logic.

```jsx
<>
  <Stepper.ActionRow>
    <Button onClick={handleHighlightTitleNext}>Next</Button>
  </Stepper.ActionRow>

  <Stepper.ActionRow>
    <Button onClick={handleSelectContentNext}>Next</Button>
  </Stepper.ActionRow>
</>
```

If there is logic shared between the two "Next" events, maybe just the share bit itself can be abstracted out into a helper function.